erpnext/public/js/utils/serial_no_batch_selector.js

```diff
@@ -0,0 +1,349 @@
+			},
+			{fieldtype:'Column Break'},
+			{
+				fieldname: 'qty',
+				fieldtype:'Float',
+				read_only: 1,
+				label: __(me.has_batch ? 'Total Qty' : 'Qty'),
+				default: me.qty
+			},
+		];
+
+		if(this.has_batch) {
+			title = __("Select Batch Numbers");
+			fields = fields.concat(this.get_batch_fields());
+		} else {
+			title = __("Select Serial Numbers");
+			fields = fields.concat(this.get_serial_no_fields());
+		}
+
+		this.dialog = new frappe.ui.Dialog({
+			title: title,
+			fields: fields
+		});
+
+		this.bind_qty();
+
+		this.dialog.set_primary_action(__('Insert'), function() {
+			me.values = me.dialog.get_values();
+			if(me.validate()) {
+				me.set_items();
+				me.dialog.hide();
+			}
+		});
+
+		this.dialog.show();
+	},
+
+	validate: function() {
+		let values = this.values;
+		if(!values.warehouse) {
+			frappe.throw(__("Please select a warehouse"));
+			return false;
+		}
+		if(this.has_batch) {
+			if(values.batches.length === 0 || !values.batches) {
+				frappe.throw(__("Please select batches for batched item "
+					+ values.item_code));
+				return false;
+			}
+			values.batches.map((batch, i) => {
+				if(!batch.selected_qty || batch.selected_qty === 0 ) {
+					frappe.throw(__("Please select quantity on row " + (i+1)));
+					return false;
+				}
+			});
+			return true;
+
+		} else {
+			let serial_nos = values.serial_no || '';
+			if (!serial_nos || !serial_nos.replace(/\s/g, '').length) {
+				frappe.throw(__("Please enter serial numbers for serialized item "
+					+ values.item_code));
+				return false;
+			}
+			return true;
+		}
+	},
+
+	set_items: function() {
+		var me = this;
+		if(this.has_batch) {
+			this.values.batches.map((batch, i) => {
+				let item_code_field = {};
+				let row = (i !== 0) ? this.frm.add_child("items") : this.item;
+				refresh_field("items");
+				if(i !== 0) {
+					let index = i + me.item.idx - 1;
+					let grid_row = me.frm.fields_dict.items.grid.grid_rows[index];
+					grid_row.toggle_editable_row();
+					item_code_field = grid_row.on_grid_fields_dict.item_code;
+
+					row.item_code = this.item_code;
+					this.map_row_values(row, batch, 'batch_no',
+						'selected_qty', this.values.warehouse, item_code_field);
+				} else {
+					this.map_row_values(row, batch, 'batch_no',
+					'selected_qty', this.values.warehouse);
+				}
+			});
+		} else {
+			this.map_row_values(this.item, this.values, 'serial_no', 'qty');
+		}
+		refresh_field("items");
+	},
+
+	map_row_values: function(row, values, number, qty_field, warehouse, item_code_field) {
+		row.qty = values[qty_field];
+		row[number] = values[number];
+		if(this.warehouse_details.type === 'Source Warehouse') {
+			row.s_warehouse = values.warehouse || warehouse;
+		} else if(this.warehouse_details.type === 'Target Warehouse') {
+			row.t_warehouse = values.warehouse || warehouse;
+		} else {
+			row.warehouse = values.warehouse || warehouse;
+		}
+
+		if(item_code_field) {
+			item_code_field.set_value(this.item_code);
```

Comment:
So yes, `set_value()`, the only way one trigger the remaining fields.
Seems it still has a problem, completely setting only the last row in the group (the first row is already set initially), leaving all the others partially filled:
![set_value_miss](https://user-images.githubusercontent.com/5196925/27571341-a3bd4c96-5b22-11e7-9f63-d3c63aa102b8.gif)
This probably has to do with it being called in bulk for all the items.